oblt-cli/cluster-create-serverless/action.yml

```diff
@@ -0,0 +1,96 @@
+    description: 'The project type. Default: `observability`'
+    default: "observability"
+    required: false
+  github-token:
+    description: 'The GitHub access token.'
+    required: true
+  slack-channel:
+    description: 'The slack channel to notify the status.'
+    default: '#observablt-bots'
+    required: false
+  username:
+    description: 'Username to show in the deployments with oblt-cli, format: [a-z0-9]'
+    default: 'obltmachine'
+    required: false
+  gitops:
+    description: 'Whether to provide the GitOps metadata to the oblt-cli'
+    default: 'false'
+    required: false
+  dry-run:
+    description: 'Whether to dryRun'
+    default: 'false'
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Configure oblt-cli flags
+      uses: actions/github-script@v7
+      with:
+        script: |
+          console.log(`Parsing flags`)
+          const { cluster_name_prefix, cluster_name_suffix, dry_run, gitops, target, project_type  } = process.env
+          if (cluster_name_prefix != '') {
+            core.exportVariable('CLUSTER_NAME_PREFIX', `--cluster-name-prefix=${cluster_name_prefix}`)
+          }
+          if (cluster_name_suffix != '') {
+            core.exportVariable('CLUSTER_NAME_SUFFIX', `--cluster-name-suffix=${cluster_name_suffix}`)
+          }
+          if (dry_run != 'false') {
+            core.exportVariable('DRY_RUN', `--dry-run`)
+          }
+          var parameters = {}
+          parameters.ProjectType = project_type
+          parameters.Target = target
+
+          if (gitops == 'true') {
+            console.log(`Parsing gitops flags`)
+            parameters.GitOps = "true"
+            parameters.GitHubRepository = `${context.repo.owner}/${context.repo.repo}`
+            if (context.eventName == 'issues') {
+              if (context.issue.number) {
+                parameters.GitHubIssue = context.issue.number
+              }
+            } else {
+              parameters.GitHubPullRequest = context.issue.number
+            }
+          }
+          parameters.EphemeralCluster = "true"
+          console.log(`Create PARAMETERS env variable`)
+          core.exportVariable('PARAMETERS', `--parameters='${JSON.stringify(parameters)}'`)
+      env:
+        cluster_name_prefix: ${{ inputs.cluster-name-prefix }}
+        cluster_name_suffix: ${{ inputs.cluster-name-suffix }}
+        dry_run: ${{ inputs.dry-run }}
+        gitops: ${{ inputs.gitops }}
+        target: ${{ inputs.target }}
+        project_type: ${{ inputs.project-type }}
+
+    - uses: elastic/oblt-actions/oblt-cli/setup@v1.2.0
+      with:
+        github-token: ${{ inputs.github-token }}
+        slack-channel: ${{ inputs.slack-channel }}
+        username: ${{ inputs.username }}
+    - run: |
+        oblt-cli cluster create custom \
+          --template serverless \
+          $DRY_RUN \
+          $CLUSTER_NAME_PREFIX \
+          $CLUSTER_NAME_SUFFIX \
+          $(echo -n $PARAMETERS | xargs)
```

Comment:
what's the reason?

Reply:
Added comment in https://github.com/elastic/oblt-actions/pull/13/commits/614869bb173785ee02d0b4f8b045d67edf46ddb9

generally, I wanted to avoid injection possibilities. although I'm not 100% sure if this is safe either

Reply:
> The best practice to avoid code and command injection vulnerabilities in GitHub workflows is to set the untrusted input value of the expression to an intermediate environment variable:

See https://securitylab.github.com/research/github-actions-untrusted-input/

Maybe we can use it?

Reply:
But thinking of it now. Maybe it's not necessary. The PARAMETERS env is not an untrusted input